File: php/features.php
<?php
declare(strict_types=1);

// Pokemon3d SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class Pokemon3dFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new Pokemon3dBaseFeature();
            case "test":
                return new Pokemon3dTestFeature();
            default:
                return new Pokemon3dBaseFeature();
        }
    }
}

File: py/features.py
# Pokemon3d SDK feature factory

from feature.base_feature import Pokemon3dBaseFeature
from feature.test_feature import Pokemon3dTestFeature


def _make_feature(name):
    features = {
        "base": lambda: Pokemon3dBaseFeature(),
        "test": lambda: Pokemon3dTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
